Mathlib/GroupTheory/Subsemigroup/Center.lean

```diff
@@ -181,22 +295,31 @@ def center : Subsemigroup M where
 
 variable {M}
 
+/-- The center of a magma is commutative and associative. -/
+@[to_additive "The center of an additive magma is commutative and associative."]
+instance [Mul M] : CommSemigroup (center M) where
+  mul_assoc _ b _ := Subtype.ext <| b.2.mid_assoc _ _
+  mul_comm a _ := Subtype.ext <| a.2.comm _
+
+variable [Semigroup M]
+
 @[to_additive]
-theorem mem_center_iff {z : M} : z ∈ center M ↔ ∀ g, g * z = z * g :=
-  Iff.rfl
+theorem mem_center_iff {z : M} : z ∈ center M ↔ ∀ g, g * z = z * g := by
+  rw [← Semigroup.mem_center_iff]
+  exact Iff.rfl
 #align subsemigroup.mem_center_iff Subsemigroup.mem_center_iff
 #align add_subsemigroup.mem_center_iff AddSubsemigroup.mem_center_iff
 
 @[to_additive]
 instance decidableMemCenter (a) [Decidable <| ∀ b : M, b * a = a * b] : Decidable (a ∈ center M) :=
-  decidable_of_iff' _ mem_center_iff
+  decidable_of_iff' _ Semigroup.mem_center_iff
 #align subsemigroup.decidable_mem_center Subsemigroup.decidableMemCenter
 #align add_subsemigroup.decidable_mem_center AddSubsemigroup.decidableMemCenter
 
 /-- The center of a semigroup is commutative. -/
 @[to_additive "The center of an additive semigroup is commutative."]
 instance center.commSemigroup : CommSemigroup (center M) :=
-  { MulMemClass.toSemigroup (center M) with mul_comm := fun _ b => Subtype.ext <| b.2 _ }
+  { MulMemClass.toSemigroup (center M) with mul_comm := fun a _ => Subtype.ext <| a.2.comm _ }
```

Comment:
This instance is a duplicate of the one above, right? I think it can be deleted.

Reply:
Yes, I think you're right. Done.